dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/dispatcher/internal/channel/HttpDispatcherLink.java

```diff
@@ -193,24 +194,43 @@ public void close(VirtualConnection conn, Exception e) {
         // so we will have to use close API from SRTConnectionContext31 and call closeStreams.
         String closeNonUpgraded = (String) (this.vc.getStateMap().get(TransportConstants.CLOSE_NON_UPGRADED_STREAMS));
         if (closeNonUpgraded != null && closeNonUpgraded.equalsIgnoreCase("true")) {
-
             if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "close streams from HttpDispatcherLink.close");
+                Tr.debug(tc, "close, CLOSE_NON_UPGRADED_STREAMS");
+            }
+
+            // Save the remain upgrading and unread data into a VC's stateMap which will be consumed in the UpgradeInputByteBufferUtil.initialRead
```

Comment:
Would this be better to just say "Save the remaining unread data"? I'm not following what upgrading vs unread data is.